qiskit_experiments/randomized_benchmarking/rb_experiment.py

```diff
@@ -0,0 +1,131 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Standard RB Experiment class.
+"""
+from typing import Union, Iterable, Optional
+
+import numpy as np
+from numpy.random import Generator, default_rng
+
+from qiskit import QuantumCircuit
+from qiskit.quantum_info import Clifford, random_clifford
+
+from qiskit_experiments.base_experiment import BaseExperiment
+from .rb_analysis import RBAnalysis
+
+
+class RBExperiment(BaseExperiment):
+    """RB Experiment class"""
+
+    # Analysis class for experiment
+    __analysis_class__ = RBAnalysis
+
+    def __init__(
+        self,
+        qubits: Union[int, Iterable[int]],
+        lengths: Iterable[int],
+        num_samples: int = 1,
+        seed: Optional[Union[int, Generator]] = None,
+        full_sampling: bool = False,
+    ):
+        """Standard randomized benchmarking experiment
+        Args:
+            qubits: the number of qubits or list of
+                    physical qubits for the experiment.
+            lengths: A list of RB sequences lengths.
+            num_samples: number of samples to generate for each
+                         sequence length
+            seed: Seed or generator object for random number
+                  generation. If None default_rng will be used.
+            full_sampling: If True all Cliffords are independently sampled for
+                           all lengths. If False for sample of lengths longer
+                           sequences are constructed by appending additional
+                           Clifford samples to shorter sequences.
+        """
+        if not isinstance(seed, Generator):
+            self._rng = default_rng(seed=seed)
+        else:
+            self._rng = seed
```

Comment:
Maybe we could have in `BaseExperiment` something general to take care of the random seed. For example have the constructor of `BaseExperiment` take `seed` as an optional parameter, and move these lines there.

Reply:
Good idea.